src/vscode-bicep/src/visualizer/view.ts

```diff
@@ -0,0 +1,211 @@
+      BicepVisualizerView.viewType,
+      visualizerTitle,
+      viewColumn,
+      {
+        enableScripts: true,
+        retainContextWhenHidden: true,
+      }
+    );
+
+    return new BicepVisualizerView(
+      languageClient,
+      webviewPanel,
+      extensionUri,
+      documentUri
+    );
+  }
+
+  public static revive(
+    languageClient: LanguageClient,
+    webviewPanel: vscode.WebviewPanel,
+    extensionUri: vscode.Uri,
+    documentUri: vscode.Uri
+  ): BicepVisualizerView {
+    return new BicepVisualizerView(
+      languageClient,
+      webviewPanel,
+      extensionUri,
+      documentUri
+    );
+  }
+
+  public reveal(): void {
+    this.webviewPanel.reveal();
+  }
+
+  public dispose(): void {
+    super.dispose();
+
+    this.webviewPanel.dispose();
+
+    // Final cleanup.
+    this.onDidDisposeEmitter.fire();
+    this.onDidDisposeEmitter.dispose();
+  }
+
+  // Do "fire and forget" since there's no need to wait on rendering.
+  public render = debounce(() => this.doRender());
+
+  private async doRender() {
+    if (this.isDisposed || !this.readyForRender) {
+      return;
+    }
+
+    let document: vscode.TextDocument;
+    try {
+      document = await vscode.workspace.openTextDocument(this.documentUri);
+    } catch {
+      this.webviewPanel.webview.html = this.createDocumentNotFoundHtml();
+      return;
+    }
+
+    if (this.isDisposed) {
+      return;
+    }
+
+    const deploymentGraph = await this.languageClient.sendRequest(
+      deploymentGraphRequestType,
+      {
+        textDocument: this.languageClient.code2ProtocolConverter.asTextDocumentIdentifier(
+          document
+        ),
+      }
+    );
+
+    if (this.isDisposed) {
+      return;
+    }
+
+    this.webviewPanel.webview.postMessage(
+      createDeploymentGraphMessage(this.documentUri.fsPath, deploymentGraph)
+    );
+  }
+
+  private handleDidReceiveMessage(message: Message): void {
+    if (message.kind === "READY") {
+      this.readyForRender = true;
+      this.render();
+    }
+  }
+
+  private createWebviewHtml() {
+    const { cspSource } = this.webviewPanel.webview;
+    const nonce = BicepVisualizerView.getNonce();
+    const scriptUri = this.webviewPanel.webview.asWebviewUri(
+      vscode.Uri.joinPath(this.extensionUri, "out", "visualizer.js")
+    );
+
+    return `
+      <!DOCTYPE html>
+      <html lang="en">
+      <head>
+        <meta charset="UTF-8">
+        <!--
+        Use a content security policy to only allow loading images from https or from our extension directory,
+        and only allow scripts that have a specific nonce.
+        -->
+        <meta http-equiv="Content-Security-Policy" content="default-src 'none'; style-src ${cspSource} 'unsafe-inline'; img-src ${cspSource} https: data:; script-src 'nonce-${nonce}' vscode-webview-resource:;">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <title>Cat Scratch</title>
```

Comment:
?

Reply:
Ahh good catch. This was the from the VSCode webview example. Will remove it.